tests/sentry/tasks/test_relocation.py

```diff
@@ -1099,3 +1112,70 @@ def test_success(self, completed_mock: Mock, fake_kms_client: FakeKeyManagementS
         # TODO(getsentry/team-ospo#203): Should notify users instead.
         assert completed_mock.call_count == 1
         assert Organization.objects.filter(slug__startswith="testing").count() == org_count + 1
+
+
+@patch(
+    "sentry.backup.helpers.KeyManagementServiceClient",
+    new_callable=lambda: FakeKeyManagementServiceClient,
+)
+@patch(
+    "sentry.tasks.relocation.CloudBuildClient",
+    new_callable=lambda: FakeCloudBuildClient,
+)
+@region_silo_test
+class EndToEndTest(RelocationTaskTestCase, TransactionTestCase):
+    def setUp(self):
+        RelocationTaskTestCase.setUp(self)
+        TransactionTestCase.setUp(self)
+
+        self.storage = get_storage()
+        files = [
+            "null.json",
+            "import-baseline-config.json",
+            "import-colliding-users.json",
+            "import-raw-relocation-data.json",
+            "export-baseline-config.json",
+            "export-colliding-users.json",
+            "export-raw-relocation-data.json",
+            "compare-baseline-config.json",
+            "compare-colliding-users.json",
+        ]
+        for file in files:
+            self.storage.save(
+                f"relocations/runs/{self.relocation.uuid}/findings/{file}", BytesIO(b"[]")
+            )
+
+    def test_valid_no_retries(
```

Comment:
Should we add tests for retries too?

Reply:
It turned out to be super complicated to do this, since we run async tasks on the stack during testing. So our usual retry trigger (an exception) bubbles all the way up and fails the test. Didn't seem worth it to try and mock around that.